source/hic_hal/stm32/stm32h743ii/extended_features/include/DAP_vendor_ex.h

```diff
@@ -10,9 +10,10 @@ typedef enum
     ID_DAP_VendorEx36_VERSION_DETAILS,
     ID_DAP_VendorEx37_HOLD_IN_BL,
     ID_DAP_VendorEx38_RESET_DAPLINK,
+    ID_DAP_VendorEx39_READ_FLEX_HOTPLUG_ADC,
 
```

Comment:
The "FLEX_HOTPLUG" seems a little strange. How about something more descriptive of what it is, like "READ_UDC_ADAPTER_TYPE_ADC"?

I'm also wondering - is it more helpful to return an ADC value or to return the type of the connected adapter? E.g., if   ~48k is type "1" and ~55k is type "2", then maybe you could have this be READ_UDC_ADAPTER_TYPE and return 0 for nothing connected, 1 for Joker board, 2 for Spencer, etc.

I'd imagine for debugging, it might be nice to have a command for returning the ADC value, but for real use seems to me like a type might be better.